base/sparse/cholmod.jl

```diff
@@ -790,9 +838,37 @@ function sparse{Ti}(A::Sparse{Complex{Float64},Ti}) # Notice! Cannot be type sta
     end
     return convert(Hermitian{Complex{Float64},SparseMatrixCSC{Complex{Float64},Ti}}, A)
 end
-sparse(L::Factor)   = sparse(Sparse(L))
+function sparse(F::Factor)
+    s = unsafe_load(F.p)
+    if bool(s.is_ll)
+        L = Sparse(F)
+        A = sparse(L*L')
+    else
+        LD = sparse(F[:LD])
+        L, d = getLd!(LD)
+        A = scale(L, d)*L'
+    end
+    p = get_perm(F)
+    if p != [1:s.n;]
+        pinv = Array(Int, length(p))
+        for k = 1:length(p)
+            pinv[p[k]] = k
+        end
+        A = A[pinv,pinv]
```

Comment:
This is not a good idea as you also anticipated in your comment. The problem `LPx=b` has to be solved `Yy=b` first and then `Px=y`, i.e. apply the inverse permutation to the solution. The operation `A[pinv,pinv]` is probably quite expensive for a sparse matrix, but destroying the triangular structure will require a new factorization before you can solve your problem.

I can see two solutions. Either `F[:LP]` returns a `PivotedCholeskyFactor` or we only support extracting `F[:L]` and `F[:p]` and it would be the responsibility of the user to apply the permutations after the solve. I prefer the last solution for simplicity for the users as well as the maintainers.